### PR TITLE
Add Initial Model Architecture

### PR DESCRIPTION
**Pull request by ljones315**
Adds ModelActor class that is able to battle in the same way as other Actor subclasses.

The model architecture is roughly what we described in the doc:

- AbilityEmbeddings - for each ability on all 12 pokemon
  - 12 * embedding_dim output
- MoveEmbeddings - for all 4 moves on all 12 pokemon
  - 48 * embedding_dim output
- Fully Connected Layer for all numerical data (normalize when possible)
  - 7 * 6 * 2 numerical inputs into Linear layer, embedding_dim output
- Hidden layers
  - 12 + 48 + 1 input
  - embedding_dim output
- Final fully connected layer
  - embedding_dim input
  - 9 outputs, the probability distribution

Running `python player_actor.py` should start a series of battles between the model and the random player!

### CODE REVIEW

File: model/move_embedding.py [hunk 1]
```diff
@@ -10,7 +10,7 @@
 # Note that there are 901 total moves in MOVE_NAME_TO_NUM dictionary. However, the identifers 'num' do NOT
 # range from 0 to 900 as some are missing. Therefore, vocab_size will be slightly higher than actual number
 # of possible moves to allow lookup without out of index error in the Embedding.
-VOCAB_SIZE = max(MOVE_NAME_TO_NUM.values())
+VOCAB_SIZE = max(MOVE_NAME_TO_NUM.values()) + 1
```
Comment (AdavyaB7) [inline model/move_embedding.py:13]: why do we need the +1 here? Isn't the emptymove added directly to `moves.csv`?

Comment (AdavyaB7) [reply model/move_embedding.py:13]: actually never mind, max value could still be OOB if vocab size is max value.

Comment (AdavyaB7) [reply model/move_embedding.py:13]: I believe we would need similar +1 in ability embeddings too then

Comment (ljones315) [reply model/move_embedding.py:13]: yes very true, slipped my mind

File: model/model_actor.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+        action_logits = self.action_fc(processed_features)
+        action_probs = nn.Softmax(dim=-1)(action_logits)
+
+        # The output is 9 numbers which we're taking to represent the probability of choosing an action
+        # indices 0-3 represent moves 1-4 of the active pokemon
+        # indices 4-8 represent switching to pokemon 2-6
+
+        # Using topk basically just to sort the list, but we grab the original indices to store
+        choices = torch.topk(action_probs, 9).indices[0]
```
Comment (AdavyaB7) [inline model/model_actor.py:89]: Is indices[0] done because of returned tensor shape in topk.indices? I would imagine ultimately choices is indices of 9 possible actions in order from best to worst.

File: model/model_actor.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        # knowledge['opponent'].print_short_info()
+        # self.team.print_short_info()
+
+        # error handling - if we receive any error from our prev choice we move to the next best choice
+        # this isn't ideal because we can run into certain errors that wouldn't require making a new choice, or making a good choice
+        # ex. I've seen "invalid move - it's not your turn" in some of the logs, in which case we can keep the same move
+        # or trying to make a move with a fainted pokemon - we should immediately choose the best "switch" action
+        if knowledge['error']:
+            self.errors += 1
+            # using % 9 is a very hacky fix that needs to be removed.
```
Comment (AdavyaB7) [inline model/model_actor.py:42]: If errors start to wrap around, since prev_probs stays the same, why would a move or switch that previously errored not error a second time?

**Comment by AdavyaB7** [comment]
Nicely done overall! This provides us a good foundation to build on top of. I think you can merge unless you have any other changes to this PR. (side note - love the prettification done throughout haha - what package / editor did you use 👀 ).

File: model/model_actor.py (hunk 2 above)
Comment (ljones315) [reply model/model_actor.py:89]: yep. the returned tensor is of shape [1,9] (looks like Tensor([[1,2,3,4]]), so indexing at 0 gets us the 1d list (or tensor I suppose) of indices.

**Comment by ljones315** [comment]
> (side note - love the prettification done throughout haha - what package / editor did you use 👀 ).

lol I think it's autopep8 (vscode extension). I have formatOnSave turned on so it started happening by default

File: model/model_actor.py (hunk 3 above)
Comment (ljones315) [reply model/model_actor.py:42]: It only really solves for a specific error, which is what Carter and I were talking about on Teams.

For some reason sometimes we get the feedback that it's not our turn, so our submitted move gets ignored and then skipped. So the %9 comes into play if the skipped move is our only valid move, all 8 other choices are attempted before returning to the original choice.

Comment (ljones315) [reply model/model_actor.py:42]: In the long run we need to fix this error handling. Obviously because we don't want to iterate over 8 invalid moves and then break, but we also want to make sure in any case we're able to accurately choose the highest probability move.

Comment (AdavyaB7) [reply model/model_actor.py:42]: makes sense, thanks for explaining

File: model/model_actor.py (hunk 2 above)
Comment (AdavyaB7) [reply model/model_actor.py:89]: always got to play with tensor shapes
